token-group/interface/src/instruction.rs

```diff
@@ -0,0 +1,337 @@
+#[discriminator_hash_input("spl_token_group_interface:initialize_group")]
+pub struct InitializeGroup {
+    /// Update authority for the group
+    pub update_authority: OptionalNonZeroPubkey,
+    /// The maximum number of group members
+    pub max_size: u32,
+}
+
+/// Instruction data for updating the max size of a `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:update_group_max_size")]
+pub struct UpdateGroupMaxSize {
+    /// New max size for the group
+    pub max_size: u32,
+}
+
+/// Instruction data for updating the authority of a `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:update_group_authority")]
+pub struct UpdateGroupAuthority {
+    /// New authority for the group, or unset if `None`
+    pub new_authority: OptionalNonZeroPubkey,
+}
+
+/// Instruction data for initializing a new `Member` of a `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:initialize_member")]
+pub struct InitializeMember {
+    /// The pubkey of the `Group`
+    pub group: Pubkey,
+    /// The member number
+    pub member_number: u32,
+}
+
+/// All instructions that must be implemented in the SPL Token Group Interface
+///
+/// Note: Any instruction can be extended using additional required accounts by
+/// using the `InitializeExtraAccountMetaList` instruction to write
+/// configurations for extra required accounts into validation data
+/// corresponding to an instruction's unique discriminator.
+#[derive(Clone, Debug, PartialEq)]
+pub enum TokenGroupInterfaceInstruction {
+    /// Initialize a new `Group`
+    ///
+    /// Assumes one has already initialized a mint for the
+    /// group.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Group
+    ///   1. `[]`   Mint
+    ///   2. `[s]`  Mint authority
+    InitializeGroup(InitializeGroup),
+
+    /// Update the max size of a `Group`
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Group
+    ///   1. `[s]`  Update authority
+    UpdateGroupMaxSize(UpdateGroupMaxSize),
+
+    /// Update the authority of a `Group`
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Group
+    ///   1. `[s]`  Current update authority
+    UpdateGroupAuthority(UpdateGroupAuthority),
+
+    /// Initialize a new `Member` of a `Group`
+    ///
+    /// Assumes the `Group` has already been initialized,
+    /// as well as the mint for the member.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Member
+    ///   1. `[]`   Member Mint
+    ///   2. `[s]`  Member Mint authority
+    ///   3. `[w]`  Group
+    ///   4. `[]`   Group Mint
+    ///   5. `[s]`  Group Mint authority
```

Comment:
I'm thinking this should be the group update authority since the group exists. What do you think?

Reply:
I could go either way tbh. I think update authority makes more since only because you might have an NFT as the group, and the mint authority is `None`, so that's prob preferred.

Note I was also considering removing the member mint authority, wdyt?

Reply:
Hm, I'd say let's keep the member mint authority, because then someone else can include your mint in all sorts of groups without your permission, which would probably help confuse people a lot.